ENIGMAsystem/SHELL/Universal_System/Extensions/Steamworks/gameclient/gameclient.cpp

```diff
@@ -0,0 +1,176 @@
+/** Copyright (C) 2023-2024 Saif Kandil (k0T0z)
+***
+*** This file is a part of the ENIGMA Development Environment.
+***
+*** ENIGMA is free software: you can redistribute it and/or modify it under the
+*** terms of the GNU General Public License as published by the Free Software
+*** Foundation, version 3 of the license or any later version.
+***
+*** This application and its source code is distributed AS-IS, WITHOUT ANY
+*** WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+*** FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+*** details.
+***
+*** You should have received a copy of the GNU General Public License along
+*** with this code. If not, see <http://www.gnu.org/licenses/>
+**/
+
+#include "gameclient.h"
+
+#include "gc_leaderboards.h"
+#include "gc_main.h"
+#include "gc_overlay.h"
+#include "gc_remotestorage.h"
+#include "gc_statsandachievements.h"
+
+namespace steamworks_gc {
+
+// gameclient* gameclient_{nullptr};
+// gameclient* gameclient() { return gameclient_; }
+
+////////////////////////////////////////////////////////
+// Public fields & functions
+////////////////////////////////////////////////////////
+
+GameClient::GameClient()
+    : gc_overlay_(nullptr),
+      gc_statsandachievements_(nullptr),
+      gc_leaderboards_(nullptr),
+      gc_remotestorage_(nullptr),
+      steam_id_local_user_(CSteamID()),
+      steam_app_id_(INVALID_APP_ID),
+      current_game_language_(std::string()),
+      available_game_languages_(std::string()) {
+  init();
+}
+
+GameClient::~GameClient() {
+  if (nullptr != GameClient::gc_overlay_) delete GameClient::gc_overlay_;
+  if (nullptr != GameClient::gc_statsandachievements_) delete GameClient::gc_statsandachievements_;
+  if (nullptr != GameClient::gc_leaderboards_) delete GameClient::gc_leaderboards_;
+  if (nullptr != GameClient::gc_remotestorage_) delete GameClient::gc_remotestorage_;
```

Comment:
It'd be safer to use `std::unique_ptr` or other shared pointer types here if possible.

Reply:
@JoshDreamland I think it will have unnecessary overhead. Using the ``new`` keyword is countable inside the Steamworks extension and I have already used smart pointers here but rolled it back as it injects unnecessary stuff and makes the code hard to read. If Robert still wants that change, I can make it happen Just review this comment, please.

Reply:
@RobertBColton Either way, after this, this PR is officially mergeable now.